src/parser/tokens.h

```diff
@@ -0,0 +1,122 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Andrew Reynolds
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2022 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * Definitions of SMT2 tokens.
+ */
+
+#include "cvc5parser_public.h"
+
+#ifndef CVC5__PARSER__TOKENS_H
+#define CVC5__PARSER__TOKENS_H
+
+#include <sstream>
+#include <string>
+
+namespace cvc5 {
+namespace parser {
+
+/**
+ * Definitions of tokens used in Flex parsers. This must be an `int` to match
+ * Flex's auto-generated lexing code.
+ */
+enum Token : int
```

Comment:
I would make this an `enum class` and eliminate the `TOK` suffixes.

Reply:
I'd like to do this, however it seems that this would make the flex lexer specification a bit of a pain.

In particular, a case like this in the flex lexer:
```
"define-sort"   return cvc5::parser::Token::DEFINE_SORT_TOK;
```

results in the syntax error:

```
smt2/smt2_lexer.flex:103:29: error: cannot convert ‘cvc5::parser::Token’ to ‘int’ in return
```

Meaning I guess I'd have to do:
```
"define-sort"   return static_cast<int>(cvc5::parser::Token::DEFINE_SORT_TOK);
```

Then similarly, cast from `int` to `Token` in `yylex()`.

Reply:
I guess this is a tradeoff, it would be nice to have these under `Token::` officially. I'm not sure which is cleaner. WDYT?

Reply:
Oh, I see. Yeah, I think you made the right call already.